### PR TITLE
fix feature

### PR DESCRIPTION
**Pull request by muhamedsultan21**
modify function logic
modify fake data
modify test cases

### CODE REVIEW

File: src/main/kotlin/interactor/FindTheHighestDifferenceInRentalPricesInteractor.kt [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
+                val outsideCity =
+                    it.realEstatesPrices.apartmentOneBedroomOutsideOfCentre!! +
                             it.realEstatesPrices.apartment3BedroomsOutsideOfCentre!!
-                maxOf(
-                    oneBedroomDifference, threeBedroomsDifference,
-                )
+
+              abs(inCityCenter - outsideCity)
             }
         return highestDifference
```
Comment (Abdelrahman-Sedky) [inline src/main/kotlin/interactor/FindTheHighestDifferenceInRentalPricesInteractor.kt:24]: The value can be returned directly without the use of variables.

File: src/test/kotlin/interactor/FindTheHighestDifferenceInRentalPricesInteractorTest.kt [hunk 2]
```diff
@@ -1,84 +1,72 @@
...
+        assertEquals(expectedCity, actualCity)
+    } 
 
     @Test
-    fun ` should return city with highest difference price when given mixed data `() {
-        //given
-        val dataSource = FindHighestDifferenceInRentalPricesFakeData().MixedData()
-        interactor = FindTheHighestDifferenceInRentalPricesInteractor(dataSource)
-        val highestDifferenceCity = FindHighestDifferenceInRentalPricesFakeData().MixedData().getAllCitiesData()[0]
+    fun `should return city with highest difference price when given mixed data`() {
```
Comment (Abdelrahman-Sedky) [inline src/test/kotlin/interactor/FindTheHighestDifferenceInRentalPricesInteractorTest.kt:24]: Show explicitly what mixed data mean.

**Review by Abdelrahman-Sedky — CHANGES_REQUESTED** [review]
Great job so far.
Some little changes are required.

**Review by Abdelrahman-Sedky — APPROVED** [review]
LGTM

**Review by Abdelwahab-IO — APPROVED** [review]

**Pull request merged**